File: lib-base/src/main/java/com/llj/lib/base/IFragment.java
package com.llj.lib.base;

import android.arch.lifecycle.Lifecycle;
import android.support.annotation.NonNull;

/**
 * ArchitectureDemo
 * describe:
 * author liulj
 * date 2018/5/24
 */
public interface IFragment {

    void initLifecycleObserver(@NonNull Lifecycle lifecycle);
}

File: lib-base/src/main/java/com/llj/lib/base/IFragment.kt
package com.llj.lib.base

import android.arch.lifecycle.Lifecycle

/**
 * ArchitectureDemo
 * describe:
 * author liulj
 * date 2018/5/24
 */
interface IFragment :IFragmentLazy{
    fun initLifecycleObserver(lifecycle: Lifecycle)

    fun moduleName(): String {
        return "app"
    }
}
